src/vos/vos_aggregate.c

```diff
@@ -373,8 +369,14 @@ vos_agg_filter(daos_handle_t ih, vos_iter_desc_t *desc, void *cb_arg, unsigned i
 		inc_agg_counter(agg_param, desc->id_type, AGG_OP_DEL);
 		D_GOTO(out, rc = 0);
 	}
-out:
 
+	/* This MUST be the last check */
+	if (desc->id_type == VOS_ITER_OBJ && vos_bkt_iter_skip(ih, desc)) {
+		credits_consume(&agg_param->ap_credits, AGG_OP_SCAN);
```

Comment:
If there are a lot of objects to be skipped, will it exhaust the credits before arriving at the object(s) that needs aggregation?

Reply:
Yes, that's a very common case today.